sparkmagic/sparkmagic/livyclientlib/livysession.py

```diff
@@ -252,12 +260,12 @@ def sleep(self):
         sleep(self._statement_sleep_seconds)
 
     def refresh_status(self):
-        status = self._http_client.get_session(self.id)[u'state']
+        response = self._http_client.get_session(self.id)
+        status = response[u'state']
+        log_array = response[u'log']
 
-        if status in constants.POSSIBLE_SESSION_STATUS:
-            self.status = status
-        else:
-            raise LivyUnexpectedStatusException(u"Status '{}' not supported by session.".format(status))
```

Comment:
Can this be edited as part of a second PR?  I wouldn't feel comfortable making a change like this without an in-depth discussion about the tradeoffs involved, along with unit tests to go along with it